vuvuzela_test.go

```diff
@@ -0,0 +1,304 @@
+package vuvuzela
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"vuvuzela.io/vuvuzela/coordinator"
+
+	"github.com/davidlazar/go-crypto/encoding/base32"
+	"golang.org/x/crypto/ed25519"
+	"vuvuzela.io/internal/debug"
+	sharedMock "vuvuzela.io/internal/mock"
+	"vuvuzela.io/vuvuzela/internal/mock"
+
+	"vuvuzela.io/vuvuzela/convo"
+
+	"vuvuzela.io/alpenhorn/config"
+	"vuvuzela.io/alpenhorn/edtls"
+	"vuvuzela.io/alpenhorn/log"
+)
+
+type universe struct {
+	Dir            string
+	Handler        *mockHandler
+	ConfigClient   *config.Client
+	ConfigServer   *config.Server
+	ConvoConfig    *config.SignedConfig
+	GuardianPub    ed25519.PublicKey
+	GuardianPriv   ed25519.PrivateKey
+	CoordinatorPub ed25519.PublicKey
+}
+
+func (u *universe) Destroy() error {
+	// TODO close everything else
+	return os.RemoveAll(u.Dir)
+}
+
+func createVuvuzelaUniverse() *universe {
+	var err error
+
+	u := new(universe)
+
+	u.Dir, err = ioutil.TempDir("", "vuvuzela_universe_")
+	if err != nil {
+		log.Panicf("ioutil.TempDir: %s", err)
+	}
+
+	u.ConfigServer, u.ConfigClient = sharedMock.LaunchConfigServer(u.Dir)
+
+	coordinatorPub, coordinatorPriv, err := ed25519.GenerateKey(rand.Reader)
+	u.CoordinatorPub = coordinatorPub
+	if err != nil {
+		log.Panicf("ed25519.GenerateKey: %s", err)
+	}
+	u.GuardianPub, u.GuardianPriv, _ = ed25519.GenerateKey(rand.Reader)
+
+	coordinatorListener, err := edtls.Listen("tcp", "localhost:0", coordinatorPriv)
+	if err != nil {
+		log.Panicf("edtls.Listen: %s", err)
+	}
+
+	conf := &config.SignedConfig{
```

Comment:
The config server stuff takes up a bunch of room and isn't the most interesting part of the test. We should move it to a different function, or perhaps a different package.

Reply:
Moved to different function (different test)